### ISSUE TITLE
Cron stopped importing into BQ

### ISSUE THREAD

**Issue by dlorenc** [ossf/scorecard#295]
The cron stopped importing in biquery:
![image](https://user-images.githubusercontent.com/1714486/111888839-3582a000-89ae-11eb-9cd2-436a3415a0aa.png)

There's a way to debut this but I forget how right now, you need to use gcloud with the job ids to search for the errors. Opening this bug to track that so I don't forget.

**Comment by naveensrinivasan** [comment]
Wow! Is the Bug query open for everyone? We should fix this and add it to the readme.

**Comment by dlorenc** [comment]
> Wow! Is the Bug query open for everyone? We should fix this and add it to the readme.

Yup! It should be publicly readable. It's just an automated import from the GCS bucket.

**Comment by naveensrinivasan** [comment]
@dlorenc Need your help with this. I don't have permission. I am trying to update the Big Query with the latest json structure.

![image](https://user-images.githubusercontent.com/172697/113446331-9609d800-93c5-11eb-909e-1ba7bf17afaa.png)

**Comment by inferno-chromium** [comment]
@naveensrinivasan - made you bigquery admin, go ahead!

**Comment by naveensrinivasan** [comment]
> @naveensrinivasan - made you bigquery admin, go ahead!

Thanks.

**Comment by oliverchang** [comment]
This probably stopped working due to https://github.com/ossf/scorecard/commit/0eaa4ff3d0dd36cd1d200662e5fc9803cc2fd7be. The BQ import expects newline delimited JSON objects, rather than a well formed JSON file.

**Comment by dlorenc** [comment]
This is still failing. Anything I can do to help fix it?

**Comment by naveensrinivasan** [comment]
AFAIK the json dump has to be changed.

**Comment by azeemshaikh38** [comment]
Dan, see https://github.com/ossf/scorecard/issues/336. I'm working on it right now, but if the failures are blocking work in some way, I can submit a hacky quick-fix solution to unblock this. Let me know.

**Comment by oliverchang** [comment]
Unfortunately this is still failing because the Date format in the latest.json is wrong:

`"Date":"21046-04-21"`

I have a fix in #353

**Comment by oliverchang** [comment]
The last two runs have been green.

**Comment by naveensrinivasan** [comment]
The cron still isn't importing correctly for example the data for this isn't populated in this table`SELECT * FROM `openssf.scorecardcron.scorecard_latest` LIMIT 1000`

The old table isn't populated either `SELECT distinct(date) FROM `openssf.scorecardcron.scorecard` LIMIT 100` the last import in old table is `04-07`

**Comment by azeemshaikh38** [comment]
So the results are being imported - if you "ORDER BY Date DESC", you'll see the rows. However, turns out that since the output has "CheckResults" instead of "Checks", the checks in BQ are empty.

**Comment by naveensrinivasan** [comment]
> So the results are being imported - if you "ORDER BY Date DESC", you'll see the rows. However, turns out that since the output has "CheckResults" instead of "Checks", the checks in BQ are empty.

OK, Thanks! So we will need to address the `checks`/`checkresults`

**Comment by azeemshaikh38** [comment]
Yes. Working on a PR for that, will send out shortly.

**Comment by oliverchang** [comment]
Nice catch!

FYI I manually fixed up the latest.json from the last cron job and recreated the BQ table (and added Date partitionining).

**Comment by azeemshaikh38** [comment]
Awesome, thanks Oliver!
